### DISCUSSION TITLE
OID: leading . or not?

### DISCUSSION THREAD

**Discussion by LFrank2021** [opendcim/openDCIM#1562] [General]
Sorry, I have another question.
As of today I use SNMP for CDU and Sensor data.
I noticed that my newest sensor (Temp/Humidity) did not report data and the solution was to insert a leading '.' on the MIBs.
For the discovery of my CDUs I do not have the leading dot and it still works.

What is the intended way (for openDCIM)?
What is the difference (if there is one)?
Should all MIBs be entered/work the same way?

Kind regards
Frank

**Comment by wilpig** [comment]
That's just the snmp response.  we have no control over it, and we aren't really using MIBs for the translation.

**Reply by LFrank2021** [reply]
Sorry, I completely phrased this wrong.
I meant OIDs!
The OIDs I am able to set on the different templates.
In my case a template for CDUs uses OIDs without a leading dot and a template for an environmental sensor won't work without the leading dot.

**Reply by alex001x** [reply]
hello,

.1.2.354 with an initial dot: The dot before the first digit indicates that the OID is absolute, i.e., it starts at the root of the OID tree. This notation is often used in formal contexts or when it is desired to explicitly state that the OID is complete.

1.2.354 without an initial dot: Without the initial dot, the OID is generally understood to be absolute if the context does not specify otherwise. However, in some cases it could be interpreted as relative to another previously defined OID.
ex: .1.2.354 -> 1.2.354 no root

an OID starting with a dot (.1.3.6.1) clearly indicates that this is a full path from the root.


- (root)
  - 0 : ITU-T
    - ...
  - 1 : ISO
    - 0 : standard
    - 1 : registration-authority
    - 2 : member-body
      - 840 : US (États-Unis)
        - 113549 : RSA Data Security, Inc.
          - 1 : pkcs
            - 1 : pkcs-1
              - 1 : rsaEncryption
        - 10045 : ANSI X9.62 (Standards pour les courbes elliptiques)
    - 3 : identified-organization
      - ...
  - 2 : Joint ISO/ITU-T
    - ...

**Reply by LFrank2021** [reply]
Thank you for the clarification @alex001x .
Much appreciated.
